csharp/ql/src/semmle/code/csharp/Type.qll

```diff
@@ -746,7 +746,7 @@ class Class extends RefType, @class_type {
  * ```
  */
 class AnonymousClass extends Class {
-  AnonymousClass() { this.getName().matches("<%") }
+  AnonymousClass() { this.getName().matches("<>f__AnonymousType%") }
```

Comment:
We should probably extract anonymous types a bit more explicitly, such as checking `ITypeSymbol.IsAnonymousType`.

Reply:
`"<>f__AnonymousType%"` doesn't seem to be the prefix with older Roslyn versions.

Reply:
https://github.com/github/codeql/pull/4672